services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,20 +44,71 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
+    // Check if the event pointer is valid
+    if (event == NULL) {
+        return ERR_CODE_INVALID_ARG;  
+    }
+
+    // Check if the thermal manager queue handle is valid
+    if (thermalMgrQueueHandle == NULL) {
+        return ERR_CODE_INVALID_STATE;  
+    }
+
+    // Attempt to send the event to the thermal manager queue
+    if (xQueueSend(thermalMgrQueueHandle, event, THERMAL_MGR_QUEUE_TIMEOUT) == pdPASS) {
+        return ERR_CODE_SUCCESS;  
+
+    // Return error if sending the event fails
+    return ERR_CODE_QUEUE_FULL;  
+    }
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+    error_code_t errCode; 
+
+    // Create a new event
+    thermal_mgr_event_t event;
+    event.type = THERMAL_MGR_EVENT_OS_INTERRUPT; 
+
+    // Log the error if sending the event fails
+    LOG_IF_ERROR_CODE(thermalMgrSendEvent(&event)); 
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
-  while (1) {
-    
-  }
+    error_code_t errCode; 
+    lm75bd_config_t *config = (lm75bd_config_t *)pvParameters;
+
+    while (1) {
+        thermal_mgr_event_t event;
+
+        // Wait for the next event from the queue
+        if (xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdPASS) {
+            // Check if the event type is MEASURE_TEMP_CMD
+            if (event.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD) {
+                // Variable to store the current temperature
+                float temperature;
+
+                // Read the current temperature from the LM75BD sensor
+                errCode = readTempLM75BD(config->devAddr, &temperature); 
+                LOG_IF_ERROR_CODE(errCode); 
+
+                // Check for read temperature error and handle accordingly
+                RETURN_IF_ERROR_CODE(errCode);
```

Comment:
Never return from a freeRTOS task, it might cause the program to crash when run on a board. You should do if(errCode != success) continue;